samples/ChatbotExamplev2/Startup.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Azure.CognitiveServices.Personalizer;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using ChatbotSample.Bots;
+using ChatbotSample.ReinforcementLearning;
+using ChatbotSample.Services;
+using System;
+
+namespace ChatbotSample
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddControllers().AddNewtonsoftJson();
+
+            // Create the Bot Framework Adapter with error handling enabled.
+            services.AddSingleton<IBotFrameworkHttpAdapter, AdapterWithErrorHandler>();
+
+            // Create the LUIS recognizer
+            services.AddSingleton<CoffeeRecognizer>();
+
+            // Create the reinforcement learning feature manager
+            services.AddSingleton<RLContextManager>();
+
+            // Create the bot as a transient. In this case the ASP Controller is expecting an IBot.
+            services.AddTransient<IBot, Bots.PersonalizerChatbot>();
+
+            string personalizerApiKey = Configuration.GetSection("PersonalizerAPIKey").Value;
+            string personalizerEndpoint = Configuration.GetSection("PersonalizerServiceEndpoint").Value;
+            if (string.IsNullOrEmpty(personalizerEndpoint) || string.IsNullOrEmpty(personalizerApiKey))
+            {
+                throw new ArgumentException("Missing Azure Personalizer endpoint and/or API key.");
+            }
+
+            services.AddSingleton(client =>
+            {
+                return new PersonalizerClient(new ApiKeyServiceClientCredentials(personalizerApiKey))
+                {
+                    Endpoint = personalizerEndpoint,
+                };
```

Comment:
Is this where you would use the `PersonalizerCoffeeClient`? Seems like duplicated code from what is in there. Related to my other comment, a factory would work nicely here.

Reply:
Yes, I will replace this with your factory suggestion hopefully